KwartrackerTests/Mocks/MockAuthServiceClient.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  MockAuthServiceClient.swift
+//  KwartrackerTests
+//
+//  Created by John Ellie Go on 6/21/21.
+//
+
+import Foundation
+import Apollo
+@testable import Kwartracker
+
+class MockAuthServiceClient {
+    var shouldErrorRequest = false
+    var shouldLoginWithError = false
+    var shouldSignupWithError = false
+    
+    let loginResponse = SignInMutation.Data.SignInWithEmail(
+        token: "sampleToken",
+        user: SignInMutation.Data.SignInWithEmail.User(
+            id: GraphQLID(0101),
+            email: "test@test.com",
+            firstName: "John",
+            lastName: "Test",
+            gender: "M",
+            age: 18
+        )
+    )
+    
+    let registrationResponse = SignUpMutation.Data.SignUpWithEmail(token: "sampleToken")
+    
+    let loginErrorResponse = """
+        {
+          "data": {
+            "signInWithEmail": null
+          },
+          "errors": [
+            {
+              "message": "Record not found",
+              "locations": [
+                {
+                  "line": 2,
+                  "column": 5
+                }
+              ],
+              "path": [
+                "signInWithEmail"
+              ],
+              "status": "unprocessable_entity",
+              "code": 422
+            }
+          ]
+        }
+        """
+    
+    let signupErrorResponse = """
+        {
+          "data": {
+            "signUpWithEmail": null
+          },
+          "errors": [
+            {
+              "message": "Email has already been taken",
+              "locations": [
+                {
+                  "line": 2,
+                  "column": 5
+                }
+              ],
+              "path": [
+                "signUpWithEmail"
+              ],
+              "status": "unprocessable_entity",
+              "code": 422
+            }
+          ]
+        }
+        """
+    
+    private func createJSONObject(_ mockObject: String) -> JSONObject? {
+        guard let data = mockObject.data(using: .utf8) else { return nil }
+        
+        do {
+            if let jsonObject = try JSONSerialization.jsonObject(with: data,
+                                                                 options: .allowFragments) as? Dictionary<String, Any> {
```

Comment:
This is too long, and I think swiftlint will show a warning on this. Better if you do it this way. 
```
if let jsonObject = try JSONSerialization.jsonObject(
    with: data,
    options: .allowFragments
) as? Dictionary<String, Any> {
    return jsonObject
}
```